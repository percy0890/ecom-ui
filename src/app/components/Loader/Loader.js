import React from 'react';
import IMG from '../../utils/images';
import './Loader.scss';

export default function Loader() {
  return (
    <div className="loading-container">
      <img src={IMG.SPINNER} alt="loader" />
    </div>
  );
}
