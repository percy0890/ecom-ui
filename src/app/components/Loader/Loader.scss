@import '~app/styles/core.scss';

.loading-container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(255,255,255,0.4);
  z-index: 9000; // Should always be top-most element
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    width: 8rem;
    height: 8rem;
    animation: infinite-spinning 2s linear infinite;
  }

  @keyframes infinite-spinning {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
}
