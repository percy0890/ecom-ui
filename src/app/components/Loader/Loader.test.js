import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('<Loader />', () => {
  it('renders correctly', () => {
    expect(shallow(<Loader />)).toMatchSnapshot();
  });
});
