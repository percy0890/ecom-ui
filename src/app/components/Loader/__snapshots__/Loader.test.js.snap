// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders correctly 1`] = `
<div
  className="loading-container"
>
  <img
    alt="loader"
    src="IMAGE_MOCK"
  />
</div>
`;
